File: 2021/day01/go/part2/part2.go
package part2

import "strconv"

func Parse(lines []string) []int {
	var values []int
	for _, line := range lines {
		value, err := strconv.Atoi(line)
		if err != nil {
			panic(err)
		}
		values = append(values, value)
	}
	return values
}

func ComputeAnswer(lines []string) int {
	result := 0
	depths := Parse(lines)
	for index := 3; index < len(depths); index++ {
		if depths[index] > depths[index-3] {
			result += 1
		}
	}
	return result
}

File: 2015/day08/python/part2.py
from typing import List


def compute_answer(lines: List[str]) -> int:
    result = 0
    for line in lines:
        index = 0
        while index < len(line):
            # print(index, line[index])
            if line[index] == '"':
                if index == 0:
                    result += 2
                elif index == len(line) - 1:
                    result += 2
            elif line[index] == "\\":
                if line[index + 1] == "\\":
                    result += 2
                    index += 1
                elif line[index + 1] == '"':
                    result += 2
                    index += 1
                elif line[index + 1] == "x":
                    result += 1
                    index += 3
            index += 1
    return result
